<div align="center"> <b> <span style='color: #006600'> <h1> IS 362 - PROJECT 2 </h1> </span> </b> </div>
<div align="center"> <b> <span style='color: #FF0000;'> <h2> 3-1-1 Service Requests </h2> </span> </b> </div>
<br>
<div align="center"><span style="color:green"><h4> Rickardo Henry -- IS362 </h4></span></div>


The data is captured from [NYC Open Data](https://opendata.cityofnewyork.us/). We will begin by importing pandas and reading the .csv data. 

In [1]:
import pandas as pd # Importing the pandas Module For Usage.
import numpy as np

service_requests = pd.read_csv('C:/Users/rickardo.henry/Desktop/IS362_Project2/311-ServiceRequests-(200).csv')
#reading data from CSV file on Computer.
service_requests

Unique Key     Created Date      Closed Date Agency  \
0      19574549  1/11/2011 15:47   7/5/2011 16:08    TLC   
1      19579718   1/11/2011 3:00              NaN    DEP   
2      19573881  1/11/2011 11:31   7/5/2011 16:07    TLC   
3      19579917  1/11/2011 12:46              NaN    DEP   
4      19580080  1/11/2011 17:03              NaN    DEP   
5      19580187   1/11/2011 9:26              NaN   DSNY   
6      19580152  1/11/2011 20:33              NaN    DEP   
7      19580185   1/11/2011 9:25              NaN   DSNY   
8      19580189   1/11/2011 9:27              NaN   DSNY   
9      19580195  1/11/2011 13:50              NaN   DSNY   
10     19580221  1/11/2011 18:50              NaN   DSNY   
11     19580965  1/12/2011 11:55  1/20/2011 10:09    TLC   
12     19581409  1/12/2011 15:08  1/19/2011 15:27    TLC   
13     19571112  1/10/2011 11:01              NaN   DSNY   
14     19571121  1/10/2011 18:50              NaN   DSNY   
15     19571552  1/10/2011 15:14              NaN    DEP   
16     19571433  1/10/2011 12:24              NaN    DEP   
17     19571437  1/10/2011 12:36              NaN    DEP   
18     19571486  1/10/2011 14:24              NaN    DEP   
19     19571301   1/10/2011 9:11              NaN    DEP   
20     19571416  1/10/2011 11:50              NaN    DEP   
21     19571453  1/10/2011 12:58              NaN    DEP   
22     19571630  1/10/2011 18:00              NaN    DEP   
23     19571632  1/10/2011 18:05              NaN    DEP   
24     19571694  1/10/2011 21:33              NaN    DEP   
25     19571561  1/10/2011 15:18              NaN    DEP   
26     19571636  1/10/2011 18:15              NaN    DEP   
27     19571680  1/10/2011 20:51              NaN    DEP   
28     19588505  1/12/2011 19:00              NaN    DEP   
29     19588525  1/12/2011 19:44              NaN    DEP   
..          ...              ...              ...    ...   
169    19596811  1/13/2011 12:32              NaN    DEP   
170    19598689  1/14/2011 12:33  1/21/2011 11:46    TLC   
171    19605249  1/12/2011 14:45   1/14/2011 0:00    DOB   
172    19605445  1/14/2011 12:59              NaN    DEP   
173    19605463  1/14/2011 13:27              NaN    DEP   
174    19605466  1/14/2011 13:35              NaN    DEP   
175    19605106  1/14/2011 14:52              NaN   DSNY   
176    19605107  1/14/2011 15:40              NaN   DSNY   
177    19605421  1/14/2011 11:39              NaN    DEP   
178    19605423  1/14/2011 11:41              NaN    DEP   
179    19605464  1/14/2011 13:32              NaN    DEP   
180    19596454  1/13/2011 21:39              NaN   DSNY   
181    19610239  1/15/2011 11:38              NaN    DEP   
182    19610258  1/15/2011 12:20  1/15/2011 16:00    DEP   
183    19763895   2/4/2011 15:05   2/4/2011 15:29    DOF   
184    19596669   1/13/2011 6:12              NaN    DEP   
185    19596766  1/13/2011 11:18              NaN    DEP   
186    19596777  1/13/2011 11:37              NaN    DEP   
187    19596906  1/13/2011 11:44              NaN    DEP   
188    19596919  1/13/2011 16:42              NaN    DEP   
189    19596924  1/13/2011 16:52              NaN    DEP   
190    19596667   1/13/2011 5:15              NaN    DEP   
191    19596671   1/13/2011 6:27              NaN    DEP   
192    19596675   1/13/2011 6:53              NaN    DEP   
193    19606611  1/15/2011 11:01   7/5/2011 16:08    TLC   
194    19597001  1/13/2011 22:05              NaN    DEP   
195    19763896   2/4/2011 13:50   2/4/2011 15:20    DOF   
196    19596812  1/13/2011 12:31              NaN    DEP   
197    19596868  1/13/2011 14:39              NaN    DEP   
198    19596983  1/13/2011 21:23              NaN    DEP   

                                        Agency Name  \
0    Correspondence - Taxi and Limousine Commission   
1            Department of Environmental Protection   
2    Correspondence - Taxi and Limousine Commission   
3            Department of Environmental Protection   
4        

<b><span style='color: #FF0000;'> Q1. What City (borough) complained the most? </span></b>

In [7]:
borough = service_requests.groupby('Borough').count()
borough

Unique Key  Created Date  Closed Date  Agency  Agency Name  \
Borough                                                                     
BRONX                  27            27            1      27           27   
BROOKLYN               47            47            7      47           47   
MANHATTAN              64            64           12      64           64   
QUEENS                 49            49            3      49           49   
STATEN ISLAND           5             5            1       5            5   
Unspecified             7             7            6       7            7   

               Complaint Type  Descriptor  Location Type  Incident Zip  \
Borough                                                                  
BRONX                      27          27             15            25   
BROOKLYN                   47          47              2            44   
MANHATTAN                  64          64             19            62   
QUEENS                     49          49              0            47   
STATEN ISLAND               5           5              1             5   
Unspecified                 7           7              0             0   

               Incident Address    ...     Bridge Highway Name  \
Borough                            ...                           
BRONX                        26    ...                       0   
BROOKLYN                     44    ...                       0   
MANHATTAN                    54    ...                       0   
QUEENS                       48    ...                       0   
STATEN ISLAND                 4    ...                       0   
Unspecified                   0    ...                       0   

               Bridge Highway Direction  Road Ramp  Bridge Highway Segment  \
Borough                                                                      
BRONX                                 0          0                       0   
BROOKLYN                              0          0                       0   
MANHATTAN                             0          0                       0   
QUEENS                                0          0                       0   
STATEN ISLAND                         0          0                       0   
Unspecified                           0          0                       0   

               Garage Lot Name  Ferry Direction  Ferry Terminal Name  \
Borough                                                                
BRONX                        0                0                    0   
BROOKLYN                     0                0                    0   
MANHATTAN                    0                0                    0   
QUEENS                       0                0                    0   
STATEN ISLAND                0                0                    0   
Unspecified                  0                0                    0   

               Latitude  Longitude  Location  
Borough                                       
BRONX                25         25        25  
BROOKLYN             42         42        42  
MANHATTAN            62         62        62  
QUEENS               47         47        47  
STATEN ISLAND         4          4         4  
Unspecified           0          0         0  

[6 rows x 52 columns]

<span style='color: #006600;'><b>Answer Q1. We can see that the borough of Manhattan had the most complaints with a total of 64 complaints.</b></span>

<b><span style='color: #FF0000;'> Q2.  What agency received the most complaints ?</span></b>

In [3]:
agency = service_requests.groupby('Agency').count()
agency

Unique Key  Created Date  Closed Date  Agency Name  Complaint Type  \
Agency                                                                       
DEP            148           148            6          148             148   
DOB              2             2            2            2               2   
DOF              5             5            5            5               5   
DOT              4             4            4            4               4   
DSNY            25            25            0           25              25   
TLC             15            15           13           15              15   

        Descriptor  Location Type  Incident Zip  Incident Address  \
Agency                                                              
DEP            148              0           142               146   
DOB              2              0             2                 2   
DOF              5              3             3                 0   
DOT              4              0             0                 1   
DSNY            25             23            25                25   
TLC             15             11            11                 2   

        Street Name    ...     Bridge Highway Name  Bridge Highway Direction  \
Agency                 ...                                                     
DEP             146    ...                       0                         0   
DOB               2    ...                       0                         0   
DOF               0    ...                       0                         0   
DOT               1    ...                       0                         0   
DSNY             25    ...                       0                         0   
TLC               2    ...                       0                         0   

        Road Ramp  Bridge Highway Segment  Garage Lot Name  Ferry Direction  \
Agency                                                                        
DEP             0                       0                0                0   
DOB             0                       0                0                0   
DOF             0                       0                0                0   
DOT             0                       0                0                0   
DSNY            0                       0                0                0   
TLC             0                       0                0                0   

        Ferry Terminal Name  Latitude  Longitude  Location  
Agency                                                      
DEP                       0       142        142       142  
DOB                       0         2          2         2  
DOF                       0         0          0         0  
DOT                       0         0          0         0  
DSNY                      0        25         25        25  
TLC                       0        11         11        11  

[6 rows x 52 columns]

<span style='color: #006600;'><b> Answer Q2. We can see that DEP <i>(Department of Environmetal Protection)</i> received the most complaints with a total of 148 complaints.</span></b>

<span style='color: #FF0000;'><b> Q3.  What was the most common complaint type ? </b></span>

In [4]:
complaint = service_requests.groupby('Complaint Type').count()
complaint

Unique Key  Created Date  Closed Date  Agency  \
Complaint Type                                                                 
Building/Use                            1             1            1       1   
DOF Parking - Payment Issue             2             2            2       2   
DOF Property - Payment Issue            3             3            3       3   
Found Property                          2             2            0       2   
General Construction/Plumbing           1             1            1       1   
Graffiti                               25            25            0      25   
Lead                                  136           136            0     136   
Noise                                   7             7            2       7   
Sewer                                   3             3            3       3   
Street Light Condition                  2             2            2       2   
Taxi Complaint                          9             9            9       9   
Taxi Compliment                         4             4            4       4   
Traffic Signal Condition                2             2            2       2   
Water System                            2             2            1       2   

                               Agency Name  Descriptor  Location Type  \
Complaint Type                                                          
Building/Use                             1           1              0   
DOF Parking - Payment Issue              2           2              0   
DOF Property - Payment Issue             3           3              3   
Found Property                           2           2              2   
General Construction/Plumbing            1           1              0   
Graffiti                                25          25             23   
Lead                                   136         136              0   
Noise                                    7           7              0   
Sewer                                    3           3              0   
Street Light Condition                   2           2              0   
Taxi Complaint                           9           9              9   
Taxi Compliment                          4           4              0   
Traffic Signal Condition                 2           2              0   
Water System                             2           2              0   

                               Incident Zip  Incident Address  Street Name  \
Complaint Type                                                               
Building/Use                              1                 1            1   
DOF Parking - Payment Issue               0                 0            0   
DOF Property - Payment Issue              3                 0            0   
Found Property                            2                 2            2   
General Construction/Plumbing             1                 1            1   
Graffiti                                 25                25           25   
Lead                                    132               136          136   
Noise                                     6                 6            6   
Sewer                                     3                 3            3   
Street Light Condition                    0                 1            1   
Taxi Complaint                            9                 0            0   
Taxi Compliment                           0                 0            0   
Traffic Signal Condition                  0                 0            0   
Water System                              1                 1            1   

                                 ...     Bridge Highway Name  \
Complaint Type                   ...                           
Building/Use                     ...                       0   
DOF Parking - Payment Issue      ...                       0   
DOF Property - Payment Issue     ...                       0   
Found Proper

<b><span style='color: #006600;'> Answer Q3. We can see that Lead was the complaint type most common with a total of 136 complaints.</span></b>

<div align="center"><b><span style='color: #006600;'><h1> THANK YOU!! </h1></span></b></div>